### PR TITLE
fix bug in #parse code path

### PR DESCRIPTION
**Pull request by dwightguth**
So apparently using a custom parser for #parse was broken after all? This whole time? Gah.

### CODE REVIEW

**Comment by dwightguth** [comment]
Actually, don't review this yet, I want to add a test.

File: tests/regression/kast/customparser [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#!/bin/sh
+
+if [ "$1" = "foo" ]
+then
+  echo ".K"
+else
+  kast --parser ground "$@"
+fi
```
Comment (radumereuta) [inline tests/regression/kast/customparser]: I don't understand what you are trying to do here.
Again, why do you need a .sh script which works only on unix systems?

Comment (dwightguth) [inline tests/regression/kast/customparser]: You're right, it's not necessary for what I'm trying to do. Give me a sec and I'll rewrite the test without it.

**Comment by radumereuta** [comment]
This looks better. If it's done, you can merge it.

**Pull request merged**